File: src/main/java/com/neo/caption/ocr/service/CacheService.java
package com.neo.caption.ocr.service;

import com.neo.caption.ocr.constant.CacheKeyPrefix;

public interface CacheService {

    <T> T getCache(CacheKeyPrefix prefix, String key);

    <T> T putCache(CacheKeyPrefix prefix, String key, T object);

    void removeCache(CacheKeyPrefix prefix, String key);

}

File: src/main/kotlin/com/neo/caption/ocr/service/CacheService.kt
package com.neo.caption.ocr.service

import com.neo.caption.ocr.common.CommonService
import org.springframework.cache.annotation.CacheEvict
import org.springframework.cache.annotation.CachePut
import org.springframework.cache.annotation.Cacheable

@CommonService
class CacheService {

    @Cacheable(key = "#p0")
    fun <T> get(key: String, value: T): T = value

    @CachePut(key = "#p0")
    fun <T> put(key: String, value: T): T = value

    @CacheEvict(key = "#p0")
    fun evict(key: String) = Unit

}